### DISCUSSION TITLE
SFDX command prompt to turn on and configure the debug logs within Vscode

### DISCUSSION THREAD

**Discussion by jriordan22849** [forcedotcom/salesforcedx-vscode#5778] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is time consuming to setup and configure debug logs in a scratch org. It is a very repetitive process from one org to another.

**Describe the solution you'd like**
I would like a prompt where i can turn on debug logs. After the logs are enabled, I can choose the debug log level, ideally by selecting a debug log level configuration file. 
I would like to have the ability to choose the user and the period the logs should be turned on for. 
I would like to have configuration files with different log levels, within the prompt, I can choose which debug log level configuration file i want to use.

**Describe alternatives you've considered**
None

**Additional context**

**Comment by lcampos** [comment]
Thanks for opening this @jriordan22849, can you expand on why you need to have these level of granularity for debug log creation ? Do you want to use the Apex Replay debugger on different users or something unrelated to the replay debugger ?

**Comment by jriordan22849** [comment]
hey @lcampos ,
When I am doing development, I would normally set debug logs on class properties etc to print in a debug log. I would like to have the ability to easily configure the debug logs for a user and the debug log level type.  This configuration files could be maintained in the source control of the package.

This is unrelated to the replay debugger. It could be a great benefit to configure the logs from a file in Vscode instead of manually logging into a org and configuring the log for a user and the debug log levels.

Been able to configure the below in a file which can easily be transferred between orgs and configured would be of great benefit. 

<img width="1193" alt="Screenshot 2020-05-06 at 15 23 14" src="https://user-images.githubusercontent.com/6918585/81140664-88798180-8fad-11ea-9742-e2f909ddba4c.png">

Please let me know if more information is required. Thanks

**Comment by lcampos** [comment]
Ok, we'll have to think about a way of fitting this into the VSCode UX. In the meantime, have you tried using the CLI to create this records using tooling api ? All of the objects (Trace Flag, Debug Level, etc) are accessible.

cc @smaddox-sf

**Comment by jriordan22849** [comment]
hey @lcampos , is there any update in relation to this?

**Comment by smaddox-sf** [comment]
Hi @jriordan22849 - Thank you for this feature request as well. It isn't on our [near-term roadmap](https://github.com/forcedotcom/salesforcedx-vscode/wiki/Roadmap), but we'll keep the feedback in mind for the future.

**Comment by klivian** [comment]
I'd like to support this one too, if it helps move up the roadmap - not being an IdeaExchange item, can't vote on it there
